[quant] Factoring out the list of no_observers

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51169 [quant][refactor] Factor out MHA code dup from nn and nn.quantizable
* **#50459 [quant] Factoring out the list of no_observers**
* #49866 [quant] Quantizable MultiheadAttention

Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

Commit:
[quant] Factoring out the list of no_observers

Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

[ghstack-poisoned]

jerryzh168 reviewed (approved):
please address lint error as well

jerryzh168 commented:
also looks like the CI is failed

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]

Commit:
Update on "[quant] Factoring out the list of no_observers"


Some of the custom modules cannot have the observers be inserted automatically. This PR factors out that list into a separate function.

Test is not required, as it is covered by the unittests for those modules.

Differential Revision: [D26092531](https://our.internmc.facebook.com/intern/diff/D26092531)

[ghstack-poisoned]
